2단계 - 모델링 하기

jivebreaddev opened the pull request:
안녕하세요! 시훈님 이번에도 잘 부탁드리겠습니다!!


chore(Requirements Modeling): 요구사항 모델링
1. sequence diagram으로 모델링
2. 엔티티 다이어그램 추가

jivebreaddev committed:
chore(Requirements Modeling): 요구사항 모델링
1. sequence diagram으로 모델링
2. 엔티티 다이어그램 추가

sihoony commented on README.md:
```diff
@@ -188,3 +188,82 @@ docker compose -p kitchenpos up -d
+   2. `사장님`이 `메뉴 카테고리`들을 조회할 수 있다.
+   3. `사장님`이 `상품`을 생성할 수 있다.
+      - `비속어`는 포함 될 수 없다.
+      - `가격`은 양수이어야 한다.
```
양수라는 단어를 요구사항을 정리했던 단어로 변경해보면 어떨까요? 
- `상품의 가격은 0원 이상이어야 한다.`

자연스럽게 읽을 수 있을 것 같네요~! 😄

jivebreaddev replied:
`양수`를 `0원 이상`으로 표기하였습니다.

sihoony commented on README.md:
```diff
@@ -188,3 +188,82 @@ docker compose -p kitchenpos up -d
+     - `주문`을 생성할 때, `대기중` 으로 `주문 상태`를 변경한다.
+     - `메뉴` 내의 `주문상품`들이 삭제된 `상품`이면 안된다.
+     - `메뉴`가 생성된 상태이고 `전시 공개` 되어있어야한다.
+     - `메뉴 가격` 총합과 `주문`에 속한 `상품 가격` 총합이 동일해야한다.
```
총합을 대체할 수 있는 단어 무엇이 있을까요~!?

jivebreaddev replied:
`총액` 으로 변경했는데 개념적 차이가 없어보여서

질문드릴려고합니다! 

`주문 총액` 혹은 `메뉴 총액` 같은 언어로 나타내길 원하시는것일까요?

sihoony commented on README.md:
```diff
@@ -188,3 +188,82 @@ docker compose -p kitchenpos up -d
+   2. `주문 종류`: `포장주문` 
+      - 주문을 생성할 때, 주문상품들의 갯수를 감소시킬 수 없다.
+   3. `손님`이 `포장주문`을 하여 주문에 대한 메뉴를 전달 받는다.
+      - 그림의 2번: `주문`을 생성할 때, `주문 상태`는 `대기중`, `주문 종류`는 `포장 주문`으로 변경한다.(B 상태)
+      - 그림의 3번: `주문`을 수락할 때, `주문 상태`는 `접수`(C 상태)
+      - 그림의 4번: `주문`을 `손님`에게 전달할 때, `주문 상태`는 `전달`으로 변경한다.(D 상태)
+      - 그림의 5번: `주문`을 완료할 때, `주문 상태`는 `주문 완료`으로 변경한다.(E 상태)
```
상태값은 실제 상태 값/용어사전의 영문명을 사용해보면 어떨까요?
추후 모델링 기반으로 개발을 할때 상태를 쉽게 찾을 수 있겠네요~!

sihoony commented on README.md:
```diff
@@ -188,3 +188,82 @@ docker compose -p kitchenpos up -d
+     - `주문`이 생성할 때, `주문 상품`들의 `갯수`를 감소 시킬 수 없다.
+     - `주문`이 생성할 때, `주문 장소`가 필수로 기입되어야 한다.
+  3. `손님`이 `배달주문`을 하여 `손님`이 주문에 대한 메뉴를 전달 받는다.
+  - 그림의 2번: `주문`을 생성할 때, `주문 상태`는 `대기중`, `주문 종류`는 `배달 주문`으로 변경한다.(B 상태)
+  - 그림의 3번: `주문`을 수락할 때, `주문 상태`는 `접수`(C 상태)
+  - 그림의 4번: `주문`을 생성 완료할 때, `주문 상태`는 `전달`으로 변경한다.(D 상태)
+  - 그림의 5번: `주문`을 `라이더`에게 전달할 때, `주문 상태`는 `배달중`으로 변경한다.(E 상태)
+  - 그림의 6번: `주문`을 `손님`에게 전달할 때, `주문 상태`는 `배달완료`으로 변경한다.(F 상태)
+  - 그림의 7번: `주문`을 `손님`에게 전달완료할 때, `주문 상태`는 `주문완료`으로 변경한다.(G 상태)
```
- 상태값을 명확하게 작성해보는건 어떨까요?
- 한글명과 영문명을 같이 기입해보면 좋을 것 같아요!

jivebreaddev replied:
모두 변경하였습니다!

sihoony reviewed (changes requested):
안녕하세요~! 시창님,
이번 미션도 잘 진행해주셨네요 👍 
간단한 코멘트를 남겼으니 확인해주세요.
즐거운 하루 되세요. 😄

sihoony commented:
마지막 코멘트가 안남아서 추가로 남겨요~! 😭 
모델링 미션은 컨텍스트를 나눠보고 나눠진 컨텍스트의 속성과 기능(행위)로 나눠보면 좋을 것 같습니다.
요구사항과 용어사전등을 활용하여 컨텍스트 분리하여 생각하신다면 몇가지의 도메인들이 분리가 될 수 있겠네요~!
궁금하신 점은 DM이나 코멘트 남겨주세요~! 😄

Jivebread committed:
chore(Modeling): kitchenpos 모델링
- 양수 -> 0원 이상으로 변경
- 총합을 대체할만한 언어로 변경
- 상태값을 모두 영어로 표기
- 도메인 별로 분리
 - 상품
 - 주문
 - 메뉴

sihoony commented on README.md:
```diff
@@ -188,3 +188,89 @@ docker compose -p kitchenpos up -d
 ## 모델링
+![](1번요구사항.png)
+
+### 1. 상품
```
Context를 잘 분리해주셨네요! 😄 
기능에 대한 정의는 잘 정리 되었는데, 어떤 속성을 갖고 있는지도 작성해 보면 어떨까요~!?

jivebreaddev replied:
상태에 관해 추가했습니다!

sihoony reviewed (changes requested):
안녕하세요~! 시창님,
피드백 잘 반영해 주셨네요 👍 
간단한 코멘트 남겼으니 확인해주시고, 궁금하신 점은 DM이나 코멘트 남겨주세요.
즐거운 하루 되세요! 😄

Jivebread committed:
chore(Modeling): kitchenpos 모델링
- 각 도메인별 상태 정리

Jivebread committed:
chore(Modeling): kitchenpos 모델링
- 속성으로 변경

sihoony reviewed (approved):
안녕하세요~! 시창님,
피드백 잘 반영해 주셨네요! 💯 
다음 미션을 위해서 Merge하겠습니다.
즐거운 하루 되세요~ 😄
